File: 89641543-main/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>



int main(void)
{


    // Get  user input
    string userinput = get_string("Text: ");

    // length
    int i = strlen(userinput);
    int letters = 0;
    int words = 0;
    int sentences = 0;

    // Loop counting
    for (int x = 0; x < i; x++)
    {
        // Count letters
        char c = userinput[x];
        if (isalpha(c) != 0)
        {
            letters++;
        }

        // Count words
        if (c == ' ')
        {
            words++;
        }

        // Count Sentences
        if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }

    // To account for last word
    words = words + 1;

    // Calculation
    float L = ((float)letters / (float)words) * 100;
    float s = ((float)sentences / (float)words) * 100;
    float subindex = 0.0588 * L - 0.296 * s - 15.8;
    int index = round(subindex);
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}

File: 89641543-main/sentimental-readability/readability.py
from cs50 import get_string

def main():
    # Get input text from user
    text = get_string("Text: ")

    # Count letters, words, and sentences
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Calculate L and S
    L = (letters / words) * 100
    S = (sentences / words) * 100

    # Calculate Coleman-Liau index
    index = 0.0588 * L - 0.296 * S - 15.8

    # Print the grade level
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")

def count_letters(text):
    count = 0
    for char in text:
        if char.isalpha():
            count += 1
    return count

def count_words(text):
    count = 1  # Assume at least one word
    for char in text:
        if char.isspace():
            count += 1
    return count

def count_sentences(text):
    count = 0
    for char in text:
        if char in [".", "!", "?"]:
            count += 1
    return count

if __name__ == "__main__":
    main()
